internal/service/encryptionatrestprivateendpoint/pural_data_source_schema.go

```diff
@@ -0,0 +1,114 @@
+package encryptionatrestprivateendpoint
+
+import (
+	"context"
+
+	"go.mongodb.org/atlas-sdk/v20240805001/admin"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/common/conversion"
+)
+
+func PluralDataSourceSchema(ctx context.Context) schema.Schema {
+	return schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"cloud_provider": schema.StringAttribute{
+				Required:            true,
+				Description:         "Human-readable label that identifies the cloud provider for the private endpoints to return.",
+				MarkdownDescription: "Human-readable label that identifies the cloud provider for the private endpoints to return.",
+			},
+			"project_id": schema.StringAttribute{
+				Required:            true,
+				Description:         "Unique 24-hexadecimal digit string that identifies your project. Use the [/groups](#tag/Projects/operation/listProjects) endpoint to retrieve all projects to which the authenticated user has access.\n\n**NOTE**: Groups and projects are synonymous terms. Your group id is the same as your project id. For existing groups, your group/project id remains the same. The resource and corresponding endpoints use the term groups.",
+				MarkdownDescription: "Unique 24-hexadecimal digit string that identifies your project. Use the [/groups](#tag/Projects/operation/listProjects) endpoint to retrieve all projects to which the authenticated user has access.\n\n**NOTE**: Groups and projects are synonymous terms. Your group id is the same as your project id. For existing groups, your group/project id remains the same. The resource and corresponding endpoints use the term groups.",
+			},
```

Comment:
Have you seen the [`PaginatedDSSchema`](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/main/internal/common/dsschema/pagination_schema.go#L9) ? I think that could apply for this resource

Reply:
Would double check in the mms implementation if effectively pageNum and itemsPerPage are not defined, or its a miss of not adding correct annotation for api spec. Seems strange since totalCount is being returned.

Reply:
Currently not supported in the API/MMS, double-checking with the API team if this is expected.

Reply:
checked with the team, they have updated the API and those attributes are now supported so I'll update the schema accordingly.

Regarding use of `PaginatedDSSchema` I understand it may make the code a bit modular but since these schemas are being generated using scaffolding, but I don't see much value in making such changes to the generated schema and I'd like to keep the generated schema as close to what is auto-generated as possible.

If we have a strong opinion on wanting to use functions like PaginatedDSSchema, I think we should work on including it in our auto-generation workflow first.

@AgustinBettati @EspenAlbert lmk if you guys disagree

Reply:
Will update this thread after the weekly meeting today

Reply:
updated PR to not support pagination as discussed today
cc @AgustinBettati